jsapp/js/components/processing/routes.utils.ts

```diff
@@ -0,0 +1,195 @@
+// This is a collection of various utility functions related to processing
+// routes and navigation.
+
+import {generatePath} from 'react-router-dom';
+import {router} from 'js/router/legacy';
+import {ROUTES, PROCESSING_ROUTES} from 'js/router/routerConstants';
+import {getCurrentPath} from 'js/router/routerUtils';
+
+/**
+ * This is a list of available tabs in Single Processing View. Each tab uses
+ * the same string as the one being used in the matching route.
+ */
+export enum ProcessingTab {
+  Transcript = 'transcript',
+  Translations = 'translations',
+  Analysis = 'analysis',
+}
+
+const TabToRouteMap: Map<ProcessingTab, string> = new Map([
+  [ProcessingTab.Transcript, PROCESSING_ROUTES.TRANSCRIPT],
+  [ProcessingTab.Translations, PROCESSING_ROUTES.TRANSLATIONS],
+  [ProcessingTab.Analysis, PROCESSING_ROUTES.ANALYSIS],
+]);
+
+interface ProcessingRouteParts {
+  assetUid?: string;
+  qpath?: string;
+  submissionEditId?: string;
+  tab?: ProcessingTab;
+}
+
+/**
+ * For given processing path, returns all of it's params and parts. If path is
+ * not provided, function is working on current path.
+ */
+export function getProcessingRouteParts(path?: string): ProcessingRouteParts {
```

Comment:
@LMNTL TBH I don't know how I feel about this function :D It does work, and is quite packed with safeness, and has tests. But I feel that there needs to be less stupid way to do this :D

Reply:
Hmmm ... you could get the path using [URL.pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname).

For the rest of the path construction/matching, it might be best to just fall back on what React Router uses - [`matchPath()`](https://reactrouter.com/en/main/utils/match-path#matchpath) and/or [`generatePath()`](https://reactrouter.com/en/main/utils/generate-path) look like they could have promise.

Reply:
I managed to make that `matchPath` work :)